### ISSUE TITLE
Embedded relations

### ISSUE THREAD

**Issue by theofidry** [dunglas/DunglasApiBundle#53]
As explaiend [here](https://github.com/dunglas/DunglasApiBundle#normalization), given we have a relation between `User` and `Job`, it is possible to display the properties of `Job` when getting the properties of `User`, which gives the following result:

```json
{
    "@context": "/api/contexts/User",
    "@id": "/api/users/1",
    "@type": "User",
    "username": "John Doe",
    "email": "john.doe@example.com",
    "roles": [
        "ROLE_USER"
    ],
    "jobs": [
        {
            "@id": "/api/jobs/101",
            "@type": "Job",
            "title": "President",
            "abbreviation": "PR",
            "mandate": "/api/mandates/12"
        }
    ]
}
```

Now as you can see with the JSON above, I also have a relation between `Job` and `Mandate`. If I use the same pattern (adding a serialization group to `Mandate`, I get the following result:

```json
{
    "@context": "/api/contexts/User",
    "@id": "/api/users/1",
    "@type": "User",
    "username": "John Doe",
    "email": "john.doe@example.com",
    "roles": [
        "ROLE_USER"
    ],
    "jobs": [
        {
            "@id": "/api/jobs/101",
            "@type": "Job",
            "title": "President",
            "abbreviation": "PR",
            "mandate": {
                "@id": "/api/mandates/12",
                "@type": "Mandate",
                "startAt": "2016-03-14T20:34:19+01:00",
                "endAt": "2016-08-03T10:53:22+02:00",
                "jobs": [
                    "/api/jobs/32",
                    "/api/jobs/96",
                    "/api/jobs/99",
                    "/api/jobs/100",
                    "/api/jobs/101"
                ]
            }
        }
    ]
}
```

Regardless on which attribute I putted the serialization group (for ex: in this case I added it on `startAt`.

In other words, the serialization group does work for this case but completly ignore which attributes we wish to expose for the embed relation of the embed relation.

**Comment by dunglas** [comment]
Can you create a gist with your entity classes?

**Comment by theofidry** [comment]
Sure: https://gist.github.com/47f5040f01a265921048.git.

**Comment by dunglas** [comment]
Does #64 fix the issue?

**Comment by theofidry** [comment]
Problem solved thanks again and sorry for the late check!

**Comment by theofidry** [comment]
Seems like a was a bit too hasty. But it's rather a question than an issue. I get the expected behaviour with the following declaration:

```yaml
resource.job:
    parent:    api.resource
    arguments: [ ApiBundle\Entity\Job ]
    calls:
        - method:    initNormalizationContext
          arguments: [ { groups: [ user ] } ]
    tags:      [ { name: api.resource } ]

resource.mandate:
    parent:    api.resource
    arguments: [ ApiBundle\Entity\Mandate ]
    tags:      [ { name: api.resource } ]

resource.user:
    parent:    api.resource
    arguments: [ ApiBundle\Entity\User ]
    calls:
        - method:    initNormalizationContext
          arguments: [ { groups: [ user ] } ]
    tags:      [ { name: api.resource } ]
```

As you can see I added the normalization context in both `User` and `Jobs`. I that the expected behaviour?

I first thought than when you were using a context it applied on all entities, i.e. if I have a relation between `User` - `Jobs` - `Mandate` and when I get `Users` with a `user` context, it will show only the attributes on which the serialization group `user` is applied but for the three entities.

**Comment by dunglas** [comment]
It's a bug. The intended behavior is the one you described.

**Comment by sebdec** [comment]
hello
I still reproduce the issue
It seems the serialization groups in relations are not managed correctly

there is a pending fix about it ?

Thank you

**Comment by teohhanhui** [comment]
@sebdec This issue is about v1.0, which means it's really old and lots of things have changed since then. Please open a new issue (if there is not already one).
